File: Gugudan.java
import java.util.Scanner;

public class Gugudan {
    private static void printGugudan(int dan) {
        for (int i = 1; i <= 9; i++) {
            System.out.printf("%d x %d = %d%n", dan, i, dan * i);
        }
    }

    private static void printAllGugudan() {
        for (int dan = 2; dan <= 9; dan++) {
            System.out.printf("%n[%d단]%n", dan);
            printGugudan(dan);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("출력할 단을 입력하세요(2~9, 전체는 Enter): ");
        String input = scanner.nextLine().trim();

        if (input.isEmpty()) {
            printAllGugudan();
            return;
        }

        try {
            int dan = Integer.parseInt(input);
            if (dan < 2 || dan > 9) {
                System.out.println("2부터 9 사이의 숫자를 입력하세요.");
                return;
            }
            printGugudan(dan);
        } catch (NumberFormatException e) {
            System.out.println("숫자를 입력하세요.");
        }
    }
}

File: gugudan.py
def print_gugudan(dan):
    for i in range(1, 10):
        print(f"{dan} x {i} = {dan * i}")


def print_all_gugudan():
    for dan in range(2, 10):
        print(f"\n[{dan}단]")
        print_gugudan(dan)


def main():
    value = input("출력할 단을 입력하세요(2~9, 전체는 Enter): ").strip()

    if not value:
        print_all_gugudan()
        return

    if not value.isdigit():
        print("숫자를 입력하세요.")
        return

    dan = int(value)
    if dan < 2 or dan > 9:
        print("2부터 9 사이의 숫자를 입력하세요.")
        return

    print_gugudan(dan)


if __name__ == "__main__":
    main()
